### PR TITLE
11.0 mig partner phone extension

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
Please remove the starting `# -*- coding: utf-8 -*-` lines. Also, could you have a look to the runbot log? There are a pair of warnings.

**Comment by JayVora-SerpentCS** [comment]
@lreficent Did the needful, thanks. BTW, the hasClass was a good change had to be done in Kanban.Thanks.

File: partner_phone_extension/views/res_partner_view.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            </field>
+        </field>
+    </record>
+
+    <record id="view_partner_number_extension_form" model="ir.ui.view">
+        <field name="name">view.partner.number.extension.form</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="/form/sheet/group/group/field[@name='phone']" position="after">
```
Comment (LoisRForgeFlow) [inline partner_phone_extension/views/res_partner_view.xml]: This gives an error:

`Element '<xpath expr="/form/sheet/group/group/field[@name='phone']">' cannot be located in parent view`

see: https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_partner_view.xml#L179

there are 2 groups inside the first one. Try: `/form/sheet/group/group[2]/field[@name='phone']`

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]

File: partner_phone_extension/views/res_partner_view.xml (hunk 1 above)
Comment (JayVora-SerpentCS) [reply partner_phone_extension/views/res_partner_view.xml]: It is tricky in odoo 11 that if you install the module alone, you will not get an error. You install SMS, it spoils the view. You install sales, it spoils the view. Logically, there is no dependency on sale btw.

**Comment by LoisRForgeFlow** [comment]
@JayVora-SerpentCS still failling on runbot

**Comment by JayVora-SerpentCS** [comment]
Works good!

**Comment by LoisRForgeFlow** [comment]
@JayVora-SerpentCS No, it doesn't:

1. Go to runbot
2. Install sales
3. Error
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/models.py", line 967, in _validate_fields
    check(self)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 333, in _check_xml
    view_def = view.read_combined(['arch'])
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 717, in read_combined
    arch = self.apply_view_inheritance(arch_tree, root.id, self.model)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 666, in apply_view_inheritance
    source = self.apply_inheritance_specs(source, specs_tree, view_id)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 644, in apply_inheritance_specs
    self.raise_view_error(_("Element '%s' cannot be located in parent view") % tag, inherit_id)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 506, in raise_view_error
    raise ValueError(message)
ValueError: Element '<xpath expr="/form/sheet/group/group[2]/field[@name='phone']">' cannot be located in parent view

Error context:
View `view.partner.number.extension.form`
[view_id: 235, xml_id: partner_phone_extension.view_partner_number_extension_form, model: res.partner, parent_id: 112]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 401, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/.repo_requirements/odoo/odoo/models.py", line 3286, in create
    self._fields[key].determine_inverse(record)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1081, in determine_inverse
    getattr(records, self.inverse)()
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 269, in _inverse_arch
    view.write(data)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 417, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/.repo_requirements/odoo/odoo/models.py", line 3008, in write
    self._write(old_vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 3161, in _write
    self._validate_fields(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 971, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nElement \'<xpath expr="/form/sheet/group/group[2]/field[@name=\'phone\']">\' cannot be located in parent view\n\nError context:\nView `view.partner.number.extension.form`\n[view_id: 235, xml_id: partner_phone_extension.view_partner_number_extension_form, model: res.partner, parent_id: 112]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 934, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 922, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-39>", line 2, in button_immediate_install
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 448, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 541, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/.repo_requirements/odoo/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 343, in load_modules
    loaded_modules, update_module)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 242, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 156, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 94, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 788, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 849, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 748, in parse
    exc_info[2]
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 401, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/.repo_requirements/odoo/odoo/models.py", line 3286, in create
    self._fields[key].determine_inverse(record)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1081, in determine_inverse
    getattr(records, self.inverse)()
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 269, in _inverse_arch
    view.write(data)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 417, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/.repo_requirements/odoo/odoo/models.py", line 3008, in write
    self._write(old_vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 3161, in _write
    self._validate_fields(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 971, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Element '<xpath expr="/form/sheet/group/group[2]/field[@name='phone']">' cannot be located in parent view

Error context:
View `view.partner.number.extension.form`
[view_id: 235, xml_id: partner_phone_extension.view_partner_number_extension_form, model: res.partner, parent_id: 112]
None" while parsing /home/odoo/OCB-11.0/addons/sms/views/res_partner_views.xml:5, near
<record id="partner_form_send_sms_form_view" model="ir.ui.view">
        <field name="name">res.partner.form.send.sms</field>
        <field name="model">res.partner</field>
        <field name="inherit_id" ref="base.view_partner_form"/>
        <field name="priority">10</field>
        <field name="arch" type="xml">
            <data><xpath expr="//field[@name='phone']" position="replace">
                <label for="phone"/>
                <div class="o_row">
                    <field name="phone" widget="phone"/>
                    <button type="action" name="%(sms.send_sms_form_action)d" class="btn-xs btn-link mb4 fa fa-envelope-o" attrs="{'invisible':[('phone', '=', False)]}" context="{'field_name': 'phone'}"/>
                </div>
            </xpath>
            <xpath expr="//field[@name='mobile']" position="replace">
                <label for="mobile"/>
                <div class="o_row">
                    <field name="mobile" widget="phone"/>
                    <button type="action" name="%(sms.send_sms_form_action)d" class="btn-xs btn-link mb4 fa fa-envelope-o" attrs="{'invisible':[('mobile', '=', False)]}" context="{'field_name': 'mobile'}"/>
                    </div>
                </xpath>
        </data></field>
    </record>
```

**Comment by JayVora-SerpentCS** [comment]
:) As said, the multiple modules make it inconsistent. I already pointed that thing. What we can do is, depend on SMS module and  handle this fix. I am sad that for the simple module, we had to do this much due to module incosistency.

**Comment by LoisRForgeFlow** [comment]
@JayVora-SerpentCS where did you point it before? anyway, these inconsistencies need to be fixed, we don't want inconsistent modules merged in OCA repositories :disappointed:

**Comment by JayVora-SerpentCS** [comment]
True, we will work this out smartly making sure it does not break us. Problem of odoo makes us take long time now:(

**Comment by JayVora-SerpentCS** [comment]
@lreficent Lets get this going. Thanks. Have to install Sales and then contacts

File: partner_phone_extension/README.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/134/11.0
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/operating-unit/issues>`_. In case of trouble, please
```
Comment (serpentcs-dev1) [inline partner_phone_extension/README.rst]: Please change it to : <https://github.com/OCA/partner-contact/issues>

**Review by serpentcs-dev1 — COMMENTED** [review]
Code Review LGTM :+1: 

Functional Review Pending!
(I rebuilt the runbot, but still does not seems to work )

**Comment by JayVora-SerpentCS** [comment]
(y) thanks.

**Review by hieulucky111 — APPROVED** [review]

**Review by novawish — APPROVED** [review]

**Pull request merged**